### PR TITLE
godoc: clarify WithTimeout deprecation note

### PR DESCRIPTION
**Pull request by ensonic**
Tell people to replace Dial + WithTimeout by DialContext + context.WithTimeout.

### CODE REVIEW

**Comment by ensonic** [comment]
FYI, the travis failure looks unrelated to me, but I don't see how I can ask it to rerun.

File: dialoptions.go [hunk 1]
```diff
@@ -374,8 +374,8 @@ func init() {
 // is returned by f, gRPC checks the error's Temporary() method to decide if it
 // should try to reconnect to the network address.
 //
-// Deprecated: use WithContextDialer instead.  Will be supported throughout
-// 1.x.
+// Deprecated: use WithContextDialer or DialContext instead of Dial. Will be
```
Comment (dfawley) [inline dialoptions.go]: This is incorrect.  `WithContextDialer` is to be used instead of `WithDialer` itself.  I don't think any change here is necessary.

Comment (ensonic) [reply dialoptions.go]: You are right. I finally found a working way to port from the deprecated api.

File: dialoptions.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ type dialOptions struct {
 	chainUnaryInts  []UnaryClientInterceptor
 	chainStreamInts []StreamClientInterceptor
 
-	cp          Compressor
+	cp          Compressor* t
```
Comment (dfawley) [inline dialoptions.go]: Please remove

File: dialoptions.go [hunk 3]
```diff
@@ -374,8 +374,7 @@ func init() {
 // is returned by f, gRPC checks the error's Temporary() method to decide if it
 // should try to reconnect to the network address.
 //
-// Deprecated: use WithContextDialer instead.  Will be supported throughout
-// 1.x.
+// Deprecated: use WithContextDialer instead. Will be supported throughout 1.x.
```
Comment (dfawley) [inline dialoptions.go]: Please re-wrap this to avoid the unnecessary diff.

File: dialoptions.go (hunk 1 above)
Comment (ensonic) [reply dialoptions.go]: Actually this is still incorrect. If you port from grpc.Dial() to grpc.DialContext you don't need WithContextDialer() at all.

Comment (dfawley) [reply dialoptions.go]: `WithContextDialer` is the new version of `WithDialer`.  `DialContext` is the new version of `Dial` (though `Dial` itself is not deprecated since a context isn't useful without `WithBlock`).  These are entirely orthogonal.

File: dialoptions.go (hunk 2 above)
Comment (ensonic) [reply dialoptions.go]: I didn't change this. I'll drop the PR:

File: dialoptions.go (hunk 3 above)
Comment (ensonic) [reply dialoptions.go]: I don't understand why you want 2 spaces separating the sentence in the docs. But anyway done.
